zebra-rpc/src/methods.rs

```diff
@@ -1255,6 +1259,29 @@ where
 
             let trees = GetBlockTrees { sapling, orchard };
 
+            let block_info_response = futs.next().await.expect("`futs` should not be empty");
+            let zebra_state::ReadResponse::BlockInfo(prev_block_info) =
+                block_info_response.map_misc_error()?
+            else {
+                unreachable!("unmatched response to a BlockInfo request");
+            };
+            let block_info_response = futs.next().await.expect("`futs` should not be empty");
+            let zebra_state::ReadResponse::BlockInfo(block_info) =
+                block_info_response.map_misc_error()?
+            else {
+                unreachable!("unmatched response to a BlockInfo request");
+            };
+
+            let delta = block_info.as_ref().and_then(|d| {
+                let value_pools = d.value_pools().constrain::<NegativeAllowed>().ok()?;
+                let prev_value_pools = prev_block_info
+                    .map(|d| d.value_pools().constrain::<NegativeAllowed>())
+                    .unwrap_or(Ok(ValueBalance::<NegativeAllowed>::zero()))
+                    .ok()?;
+                (value_pools - prev_value_pools).ok()
+            });
+            let size = size.or(block_info.as_ref().map(|d| d.size() as usize));
```

Comment:
Optional/Nitpick: 
- There doesn't seem to be any point in checking for the size from the `BlockAndSize` request anymore,
- For verbosity 2 requests, this `getblock` method could fallback on using `zcash_serialized_size()` with an `or_else()` to populate the `size` field in its output before the format upgrade is complete.

```suggestion
            let size = block_info.as_ref().map(|d| d.size() as usize);
```

Reply:
The upside to not changing this is that it keeps `size` being returned when the database upgrade is running. We could clean this up in the release after the current Zebra reaches EoS, but also not a big deal.